File: OpenSourceCode/app/src/main/java/com/tao/opensourcecode/MainActivity.java
package com.tao.opensourcecode;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import java.io.IOException;

import okhttp3.Call;
import okhttp3.Callback;
import okhttp3.MediaType;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);




    }
    public static final MediaType JSON
            = MediaType.get("application/json; charset=utf-8");

    private static final String DESTINATION_ADDRESS = "https://github.com/soulrelay";

    public void test_okhttp_sync() throws IOException {
        OkHttpClient client = new OkHttpClient.Builder().build();
        Request request = new Request.Builder()
                .url(DESTINATION_ADDRESS)
                .build();
        Response response = client.newCall(request).execute();
    }


    //private static final String DESTINATION_ADDRESS = "https://github.com/soulrelay";
    public void test_okhttp_async() {
        OkHttpClient client = new OkHttpClient.Builder().build();
        Request request = new Request.Builder()
                .url(DESTINATION_ADDRESS)
                .build();
        client.newCall(request).enqueue(new Callback() {
            @Override
            public void onFailure(Call call, IOException e) {
            }

            @Override
            public void onResponse(Call call, Response response) throws IOException {
            }
        });

    }
}

File: demo/GradleBuildDemo/app/src/main/java/com/tao/gradlebuilddemo/MainActivity.kt
package com.tao.gradlebuilddemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

class MainActivity : AppCompatActivity() {
    val tag = "MainActivity"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onStart() {
        super.onStart()
        val yunListener = object :YunListener{
            override fun onData() {

            }
        }

        setYunListener(object :YunListener{
            override fun onData() {
                Log.d(tag,"onData...")
            }
        })

        setYunListener(listener = object :YunListener{
            override fun onData() {

            }
        })
    }

    private fun setYunListener(listener: YunListener) {
        listener.onData()
    }
}